### PR TITLE
Get rid of Array.prototype/Object.prototype methods (close #245)

### LINKED ISSUE #245: Get rid of Array.prototype/Object.prototype methods in client and shared code
`Array.prototype.slice.call` is used in many places.

We need to create cloneItems(iterableObject) util and rewrite all places.
https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#32-leaking-arguments

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @inikulin @churkin

### CODE REVIEW

File: src/client/sandbox/upload/hidden-info.js [hunk 1]
```diff
@@ -31,7 +32,7 @@ export function addInputInfo (input, fileList, value) {
     if (formInfo) {
         var files = [];
 
-        Array.prototype.slice.call(fileList).forEach(file => {
+        arraySlice.call(fileList).forEach(file => {
```
Comment (inikulin) [inline src/client/sandbox/upload/hidden-info.js]: Shouldn't it warn about `forEach` as well?

File: src/utils/proto-methods.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export var arraySlice     = Array.prototype.slice;
```
Comment (inikulin) [inline src/utils/proto-methods.js]: It's assumed to be used by client, so it requires appropriate warning comment.

**Comment by inikulin** [comment]
\r-

File: src/utils/proto-methods.js (hunk 2 above)
Comment (inikulin) [inline src/utils/proto-methods.js]: We need a comment that describe why we need this stuff as well, with the link to the GH issue.

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
FPR

File: src/client/transport.js [hunk 3]
```diff
@@ -220,9 +220,13 @@ class Transport extends EventEmitter {
         var storedMessages = Transport._getStoredMessages();
 
         if (storedMessages.length) {
+            var msgLen = storedMessages.length;
+            var tasks  = new Array(msgLen);
```
Comment (inikulin) [inline src/client/transport.js]: Why not `var tasks = [];` and then `tasks.push(...)`?

File: src/utils/proto-methods.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// -------------------------------------------------------------
+// WARNING: this file is used by both the client and the server.
+// Do not use any browser or node-specific API!
+// -------------------------------------------------------------
+// GH-245
```
Comment (inikulin) [inline src/utils/proto-methods.js]: It doesn't follows our comment conventions. First of all it require `NOTE: ` prefix. Also, I've asked for the textual description with the reference to the issue.

**Comment by inikulin** [comment]
\r-

**Comment by LavrovArtem** [comment]
FPR

**Comment by LavrovArtem** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by churkin** [comment]
lgtm

**Pull request merged**